--- 20_linux_tboot	2017-08-02 16:18:41.872143371 -0400
+++ /etc/grub.d/19_linux_tboot_pretxt	2017-08-02 16:20:54.191143355 -0400
@@ -65,7 +65,7 @@
 export TEXTDOMAIN=grub
 export TEXTDOMAINDIR=${prefix}/share/locale
 
-CLASS="--class gnu-linux --class gnu --class os --class tboot"
+CLASS="--class gnu-linux --class gnu --class os --class tboot --unrestricted"
 
 if [ "x${GRUB_DISTRIBUTOR}" = "x" ] ; then
   OS=GNU/Linux
@@ -179,17 +179,6 @@
 tboot_list=`for i in /boot/tboot*.gz; do
         if grub_file_is_not_garbage "$i" ; then echo -n "$i " ; fi
       done`
-sinit_list=`for i in /boot/*sinit* /boot/*SINIT*; do
-        basename=$(basename $i)
-        if grub_file_is_not_garbage "$i" ; then echo -n "$basename " ; fi
-      done`
-if [ -n "${GRUB_TBOOT_POLICY_DATA}" ]; then
-  if grub_file_is_not_garbage "/boot/${GRUB_TBOOT_POLICY_DATA}"; then
-    poldata_file=${GRUB_TBOOT_POLICY_DATA}
-  else
-    echo "ERROR in $0: GRUB_TBOOT_POLICY_DATA=${GRUB_TBOOT_POLICY_DATA} not found in /boot, check ${sysconfdir}/default/grub-tboot" >&2
-  fi
-fi
 prepare_boot_cache=
 
 while [ "x${tboot_list}" != "x" ] && [ "x$linux_list" != "x" ] ; do
@@ -200,7 +189,7 @@
     rel_tboot_dirname=`make_system_path_relative_to_its_root $tboot_dirname`
 #    tboot_version=`echo $tboot_basename | sed -e "s,.gz$,,g;s,^tboot-,,g"`
     tboot_version="1.9.4"
-    echo "submenu \"tboot ${tboot_version}\" {"
+    echo "submenu \"tboot ${tboot_version} no policy\" --unrestricted {"
     while [ "x$list" != "x" ] ; do
 	linux=`version_find_latest $list`
 	echo "Found linux image: $linux" >&2
