--- 20_linux_tboot	2017-08-04 12:19:31.825655186 -0400
+++ /etc/grub.d/20_linux_tboot	2017-08-04 12:26:01.790655140 -0400
@@ -16,6 +16,8 @@
 #
 # You should have received a copy of the GNU General Public License
 # along with GRUB.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Modified by SIMP for fixing some tboot related bugs (tpm::tboot::grub::grub2).
 
 prefix=/usr
 exec_prefix=${prefix}
@@ -65,7 +67,7 @@
 export TEXTDOMAIN=grub
 export TEXTDOMAINDIR=${prefix}/share/locale
 
-CLASS="--class gnu-linux --class gnu --class os --class tboot"
+CLASS="--class gnu-linux --class gnu --class os --class tboot --unrestricted"
 
 if [ "x${GRUB_DISTRIBUTOR}" = "x" ] ; then
   OS=GNU/Linux
@@ -150,6 +152,13 @@
 	${mb_mod_directive} ${rel_dirname}/${initrd}
 EOF
   fi
+  if test -n "${poldata_file}" ; then
+    message="$(gettext_printf "Loading tboot policy data file ${poldata_file} ...")"
+    cat << EOF
+	echo	'$message'
+	${mb_mod_directive} ${rel_dirname}/${poldata_file}
+EOF
+  fi
   if test -n "${sinit_list}" ; then
     for i in ${sinit_list}; do
       message="$(gettext_printf "Loading sinit $i ...")"
@@ -159,13 +168,6 @@
 EOF
     done
   fi
-  if test -n "${poldata_file}" ; then
-    message="$(gettext_printf "Loading tboot policy data file ${poldata_file} ...")"
-    cat << EOF
-	echo	'$message'
-	${mb_mod_directive} ${rel_dirname}/${poldata_file}
-EOF
-  fi
   cat << EOF
 }
 EOF
@@ -200,7 +202,7 @@
     rel_tboot_dirname=`make_system_path_relative_to_its_root $tboot_dirname`
 #    tboot_version=`echo $tboot_basename | sed -e "s,.gz$,,g;s,^tboot-,,g"`
     tboot_version="1.9.4"
-    echo "submenu \"tboot ${tboot_version}\" {"
+    echo "submenu \"tboot ${tboot_version}\" --unrestricted {"
     while [ "x$list" != "x" ] ; do
 	linux=`version_find_latest $list`
 	echo "Found linux image: $linux" >&2
